### PR TITLE
LPS-130451 Discontinue "two star" wildcard search queries

### PR DESCRIPTION
**Pull request by 4lejandrito**
**Motivation**

Wildcard queries with leading and trailing wildcards are not too efficient.

**Proposed Solution**

Reuse the same search mechanism used in web content structures (or document types) search. Sadly we lose support for suffix queries.

**How to test it**

I've added a test to ensure we still support prefix search.

_To test it manually_

* Go to the content dashboard.
* Click on filter.
* Select Subtype.

Follow the steps in the ticket.

Results should be instant.

### CODE REVIEW

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:stable

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/selector/provider/ContentDashboardItemTypeItemSelectorProvider.java [hunk 1]
```diff
@@ -193,15 +175,17 @@ private SearchResponse _getSearchResponse(
...
+					searchContext.setBooleanClauses(_getBooleanClauses());
 					searchContext.setCompanyId(
 						_portal.getCompanyId(portletRequest));
 					searchContext.setEnd(end);
 					searchContext.setSorts(_getSort(portletRequest));
 					searchContext.setStart(start);
+
+					String keywords = _getKeywords(portletRequest);
+
+					searchContext.setAttribute(Field.DESCRIPTION, keywords);
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/selector/provider/ContentDashboardItemTypeItemSelectorProvider.java:187]: Maybe  setAttribute should go before  setBooleanClauses

**Review by cgoncas — APPROVED** [review]
LGTM. I have added a small comment, but I don't think it is a impediment. I have tested the PR and everything works fine.

I think that the only missing use case is search by partial words or suffixes, but This should not be an impediment either.

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**